### PR TITLE
Added Angular example

### PR DESCRIPTION
**Pull request by dxArtemiusz**
Example's Readme is in progress

### CODE REVIEW

File: angular/src/app/app.component.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    return currentTraining;
+  }
+
+
+  getColor = (id) => {
+    return this.resourcesDataSource.find((employee) => {
+      return employee.id === id;
+    }).color;
+  }
+}
```
Comment (williamvinogradov) [inline angular/src/app/app.component.ts:88]: Let's add types and make this method shorter:

```suggestion
  getColor = (id: number): string => this.resourcesDataSource.find((employee) => 
        employee.id === id
      ).color;
  }
}
```

File: angular/src/app/app.component.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    <Tooltip
+        [appointmentData]="data.appointmentData"
+        [isDeleteButtonExist]="isDeleteButtonExist(data)"
+        [markerColor]="getColor | apply : data.appointmentData.employeeID"
+        (onDeleteButtonClick)="onDeleteButtonClick($event, data.appointmentData)"
+    ></Tooltip>
+  </ng-container>
+
+  <div *dxTemplate="let employee of 'resourceCellTemplate'">
+    <div class="name" [style.background]="employee.color">
```
Comment (williamvinogradov) [inline angular/src/app/app.component.html:38]: Minor comment.

Let's move tags to the new line as we do in other places:

```suggestion
    <div 
      class="name" 
      [style.background]="employee.color"
    >
```

File: angular/src/app/app.component.html [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      <h2>{{ employee.text }}</h2>
+    </div>
+    <div class="avatar">
+      <img [src]="employee.data.avatar" />
+    </div>
+    <div class="info" [style.color]="employee.color">
```
Comment (williamvinogradov) [inline angular/src/app/app.component.html:44]: Minor comment.

Let's move tags to the new line as we do in other places:
```suggestion
    <div 
      class="info" 
      [style.color]="employee.color"
    >
```

File: angular/src/app/app.component.html [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      Age: {{ employee.data.age }}
+      <br />
+      <b>{{ employee.data.discipline }}</b>
```
Comment (williamvinogradov) [inline angular/src/app/app.component.html:47]: Let's get rid of ```<br>``` tag here:
```suggestion
      <div>Age: {{ employee.data.age }}</div>
      <div>
        <b>{{ employee.data.discipline }}</b>
      </div>
```

File: angular/src/app/app.component.html [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    </div>
+  </div>
+
+  <div
+    *dxTemplate="let cellData of 'dataCellTemplate'"
+    class="employee-weekend"
+    [ngClass]="markWeekEnd | apply : cellData"
+  >
+    <div class="day-cell" [ngClass]="markTraining | apply : cellData">
```
Comment (williamvinogradov) [inline angular/src/app/app.component.html:56]: Minor comment.

Let's move tags to the new line as we do in other places:
```suggestion
    <div 
      class="day-cell" 
      [ngClass]="markTraining | apply : cellData"
    >
```

File: angular/src/app/app.component.ts [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+import { DxSchedulerModule, DxTemplateModule, DxSchedulerComponent } from 'devextreme-angular';
+import DataSource from 'devextreme/data/data_source';
+import { Service, Employee } from './app.service';
+import { ApplyPipe } from './pipes';
+import { ClickEvent } from 'devextreme/ui/button';
+import { dxSchedulerAppointment } from 'devextreme/ui/scheduler';
+import { TooltipModule } from './tooltip/tooltip.component';
+
+if (!/localhost/.test(document.location.host)) {
```
Comment (williamvinogradov) [inline angular/src/app/app.component.ts:12]: It looks like a duplicate logic that enables production mode
Let's move it to [this place](https://github.com/DevExpress-Examples/Appointment-Template-for-DevExtreme-Scheduler/pull/1/files#diff-c5feb2cdd9864b1fa58a16d52c18253015aa8db676dc541fbc09c35f3a413de9R7) and merge these conditions :)

File: angular/src/app/app.component.ts [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    DxSchedulerModule,
+    DxTemplateModule,
+    TooltipModule
+  ],
+  declarations: [AppComponent, ApplyPipe],
+  bootstrap: [AppComponent],
+})
+export class AppModule { }
+
+platformBrowserDynamic().bootstrapModule(AppModule);
```
Comment (williamvinogradov) [inline angular/src/app/app.component.ts:102]: Let's remove this duplicate
We already has bootstrapping [here](https://github.com/DevExpress-Examples/Appointment-Template-for-DevExtreme-Scheduler/pull/1/files#diff-c5feb2cdd9864b1fa58a16d52c18253015aa8db676dc541fbc09c35f3a413de9R11)

Comment (dxArtemiusz) [reply angular/src/app/app.component.ts]: resolved in the last commit

File: angular/src/app/app.component.ts (hunk 6 above)
Comment (dxArtemiusz) [reply angular/src/app/app.component.ts]: resolved in the last commit

File: angular/src/app/tooltip/tooltip.component.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    @Input() isDeleteButtonExist: boolean;
+    @Output() onDeleteButtonClick = new EventEmitter<ClickEvent>();
+}
+
+@NgModule({
+imports: [DxButtonModule, CommonModule],
+        declarations: [TooltipComponent],
+        exports: [TooltipComponent]
+})
+export class TooltipModule { }
```
[the remark this answers is not in the document]
Comment (dxArtemiusz) [reply angular/src/app/tooltip/tooltip.component.ts]: resolved in the last commit

File: angular/src/app/app.component.ts [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  markTraining = (cellData) => {
+    const classObject = {
+      'day-cell': true,
+    };
+
+    classObject[AppComponent.getCurrentTraining(cellData.startDate.getDate(), cellData.groups.employeeID)] = true;
+    return classObject;
+  };
+
+  static getCurrentTraining(date, employeeID) {
```
[the remark this answers is not in the document]
Comment (dxArtemiusz) [reply angular/src/app/app.component.ts:71]: Resolved, but I used `date: number` as here used the `Date.getDate()` value and it is a day number instead Date.

File: angular/src/app/app.component.ts [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+          || schedulerInstance.option('editing') === true;
+
+      return !isAppointmentDisabled && isDeleteAllowed;
+  }
+
+  markWeekEnd = (cellData) => {
+    function isWeekEnd(date) {
+      const day = date.getDay();
+      return day === 0 || day === 6;
+    }
```
[the remark this answers is not in the document]
Comment (dxArtemiusz) [reply angular/src/app/app.component.ts:54]: Resolved

File: angular/src/app/app.component.ts [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+
+  getColor = (id) => {
+    return this.resourcesDataSource.find((employee) => {
+      return employee.id === id;
+    }).color;
+  }
+}
+
+@NgModule({
```
[the remark this answers is not in the document]
Comment (dxArtemiusz) [reply angular/src/app/app.component.ts]: Resolved

File: angular/src/app/tooltip/tooltip.component.ts [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    templateUrl: "./tooltip.component.html"
+})
+export class TooltipComponent {
+    @Input() appointmentData: dxSchedulerAppointment;
+    @Input() markerColor: string;
+    @Input() isDeleteButtonExist: boolean;
+    @Output() onDeleteButtonClick = new EventEmitter<ClickEvent>();
+}
+
+@NgModule({
```
[the remark this answers is not in the document]
Comment (dxArtemiusz) [reply angular/src/app/tooltip/tooltip.component.ts]: Resolved

File: angular/src/app/app.component.ts [hunk 13]
```diff
@@ -0,0 +1,82 @@
+import { Component, ViewChild } from '@angular/core';
+import { DxSchedulerComponent } from 'devextreme-angular';
+import DataSource from 'devextreme/data/data_source';
+import { Service } from './app.service';
+
+import { ClickEvent } from 'devextreme/ui/button';
+import { dxSchedulerAppointment } from 'devextreme/ui/scheduler';
+
+
+
```
Comment (williamvinogradov) [inline angular/src/app/app.component.ts:10]: Let's remove this redundant empty lines

File: angular/src/app/app.component.ts [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css'],
+  providers: [Service],
+})
+export class AppComponent {
+  @ViewChild('targetScheduler', { static: true })
+  scheduler: DxSchedulerComponent;
+  dataSource = new DataSource({
+      store: this.service.getData(),
+   });
```
Comment (williamvinogradov) [inline angular/src/app/app.component.ts:23]: Let's add empty line here, as we do it in another places:

```suggestion
  scheduler: DxSchedulerComponent;
  
  dataSource = new DataSource({
      store: this.service.getData(),
   });
```

File: angular/src/app/tooltip/tooltip.component.html [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+        class="dx-tooltip-appointment-item-delete-button-container"
+    >
+        <dx-button
+            class="dx-tooltip-appointment-item-delete-button"
+            icon="trash"
+            stylingMode="text"
+            (onClick)="onDeleteButtonClick.emit($event)"
+        ></dx-button>
+    </div>
+</div>
```
Comment (williamvinogradov) [inline angular/src/app/tooltip/tooltip.component.html:30]: Let's add a new line at the end of this file

File: angular/src/app/app.component.ts (hunk 14 above)
Comment (dxArtemiusz) [reply angular/src/app/app.component.ts]: resolved

File: angular/src/app/app.component.ts (hunk 13 above)
Comment (dxArtemiusz) [reply angular/src/app/app.component.ts]: Resolved

**Pull request merged**